src/server/pa_rot.rs

```diff
@@ -76,16 +76,21 @@ where
     ) -> Result<(), Error> {
         let result = Handler::<&mut Self>::new()
             .handle::<protocol::FirmwareVersion, _>(|zelf, req| {
-                if req.index != 0 {
-                    return Err(protocol::Error {
-                        code: protocol::ErrorCode::Unspecified,
-                        data: [0; 4],
-                    });
+                if req.index == 0 {
+                    Ok(protocol::firmware_version::FirmwareVersionResponse {
```

Comment:
Given the name is used twice, maybe `use protocol::...::Response` in the closure body?

Reply:
Done.